### PR TITLE
feat(api) : permission module added

### PR DESCRIPTION
**Pull request by soumyadip007**
Subject:  feat(api) : permission module added
Assignees: @soumyadip007 

## Closes / Fixes 

1.  New Configurations added for RBAC
2. Permission services added including Add, Delete permissions
3. Initial Owner Permissions added for the new properties

## Does this PR introduce a breaking change

NO

## Screenshot(s)

<details>
<summary>View Screenshots</summary>

</details>

### Ready-for-merge Checklist

- [ ] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?
- [ ] Does the change have appropriate unit tests?
- [ ] Did you update or add any necessary documentation (README.md, WHY.md, etc.)?
- [ ] Was this feature demo'd and the design review approved?

### CODE REVIEW

File: packages/api/src/configuration/index.ts [hunk 1]
```diff
@@ -33,6 +33,21 @@ export const AUTH_LISTING = {
...
+  const base64EncodedCreds = Buffer.from(`${ROVER_AUTH_DETAILS.username}:${ROVER_AUTH_DETAILS.password}`);
+  return base64EncodedCreds.toString('base64');
+}
+
+export const ROVER_AUTH = {
+  cred: getRoverAuth(),
+  baseUrl: process.env.SPASHIP_ROVER_BASE_URL
+};
+
 export enum MIN {
```
Comment (arkaprovob) [inline packages/api/src/configuration/index.ts:51]: The purpose of this enum is not clear from this code either rename it or add a comment

File: packages/api/src/configuration/index.ts [hunk 2]
```diff
@@ -85,3 +100,5 @@ export const VALIDATION = {
 export enum JOB {
   DELETE_EPH_ENV = 'DELETE_EPH_ENV'
```
Comment (arkaprovob) [inline packages/api/src/configuration/index.ts:101]: this is not part of RBAC, please move this to a dedicated pr

File: packages/api/src/server/permission/permission.controller.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+  @Post()
+  @ApiOperation({ description: 'Create a New Permission.' })
+  async createProperty(@Body() permissionDto: CreatePermissionDto): Promise<Permission[]> {
+    return this.permissionService.createPermission(permissionDto);
+  }
+
+  @Delete()
+  @ApiOperation({ description: 'Delete a Permission.' })
+  async deleteApiKey(@Body() deletePermissionDto: DeletePermissionDto): Promise<Permission[]> {
```
Comment (arkaprovob) [inline packages/api/src/server/permission/permission.controller.ts:35]: Rename this method, and it probably came from the copy-pasting method name.

File: packages/api/src/server/analytics/activity-stream.entity.ts [hunk 4]
```diff
@@ -36,5 +36,8 @@ export enum Action {
   ENV_SYNCED = 'ENV_SYNCED',
   WEBHOOK_CREATED = 'WEBHOOK_CREATED',
   WEBHOOK_DELETED = 'WEBHOOK_DELETED',
-  WEBHOOK_UPDATED = 'WEBHOOK_UPDATED'
+  WEBHOOK_UPDATED = 'WEBHOOK_UPDATED',
+  PERMISSION_CREATED = 'PERMISSION_CREATED',
```
Comment (arkaprovob) [inline packages/api/src/server/analytics/activity-stream.entity.ts:40]: The access actions suppose to come from a database, then what's the purpose of this enum Action?

**Review by arkaprovob — CHANGES_REQUESTED** [review]

File: packages/api/src/server/analytics/activity-stream.entity.ts (hunk 4 above)
Comment (soumyadip007) [reply packages/api/src/server/analytics/activity-stream.entity.ts:40]: This is for the Activity stream only, so we have already kept it in memory, because these are the constants. and  we'll fetch the Auth Actions details from the Database that is different.

File: packages/api/src/configuration/index.ts (hunk 1 above)
Comment (soumyadip007) [reply packages/api/src/configuration/index.ts:51]: Sure

File: packages/api/src/server/analytics/activity-stream.entity.ts (hunk 4 above)
Comment (arkaprovob) [reply packages/api/src/server/analytics/activity-stream.entity.ts:40]: Since this is not a part of RBAC, Please create a dedicated PR for this and remove this from this PR.

**Comment by soumyadip007** [comment]
Hi @arkaprovob , I've made the changes as per your request. I've added a comment for the Activity Stream specifier on Permissions. please check that, if you feel this should be moved in a Separate PR, I'll do it.

File: packages/api/src/server/analytics/activity-stream.entity.ts (hunk 4 above)
Comment (arkaprovob) [reply packages/api/src/server/analytics/activity-stream.entity.ts:40]: As discussed, I now understand the requirement, and it makes sense.

**Review by arkaprovob — APPROVED** [review]
